docs/ADRs/018-changing-form-over-time.md

```diff
@@ -0,0 +1,91 @@
+
+## Options Considered
+
+### Data Storage
+
+The first question we have to answer about changing form data is whether we want to maintain old FormDatas as they were submitted, or just migrate the data as the form changes. 
+
+#### Keep Old Submissions Intact
+
+This strategy preserves all the data we received for each package as it was when submitted. We store the data in the format it was entered in in the database and display it as it was whenever someone accesses it. 
+
+#### Migrate Submission Data
+
+This strategy would migrate old data as the shape of our form changes. If we removed a question, it would be removed from old submissions. If we add a question, a dummy answer would be added to the old submissions. This would mean that our code would only ever have to deal with the most recent version of the data, every time a submission was loaded, no matter how old, the data would come back in the same shape. 
+
+The drawback, and ultimately the deal breaker, is that we would then lose some actual answers submitted by states in the past. We decided that it was crucial to keep a complete record of what a state submitted in the past regardless of how the form has changed since. Reviews of these contracts take place on a timeline of years, we don't want to lose the record of what was submitted since that might be referenced over time in the adjudication of the submission. 
+
+
+### Schema Migration
+
+Since we want to keep old data intact and we need to be able to change the schema of the data over time, that means that our application will need to be able to deal with loading different schemas of data if it is viewing a recent or an old version of the form. 
+
+There are a variety of different changes to the form that will require changing the shape of our data. Two simple ones are adding a new field or removing an old one. More complex ones could be changing the type of a field or changing the values of an enum. We identified two broad strategies for dealing with this, one more invasive than the other. 
+
+First we can simply always add optional fields and only remove optional fields. By that I mean optional at the typescript field level. This forces our code that deals with the form data to always check on these fields and deal appropriately with the case where the data is missing. Since this mostly concerns the single page where we display the package summary, it's mostly a question of leaving question/answers off the page when they don't exist in the data. This works in both directions. Displaying an old submission that doesn't have one of the new questions on it, omit it. Displaying a new submission that doesn't have a question that has been removed, omit it. If a field's type needs to change, that can be accommodated by adding a new field and removing the old one. 
+
```

Comment:
Another drawback here we could muddy the waters for what is the source of truth for form schema required fields. Because we still do have required fields we want to return failures for in certain types of form submits (such as submitting the package overall)